Sources/Kafka/RDKafka/RDKafkaClient.swift

```diff
@@ -19,37 +19,54 @@ import Logging
 /// Base class for ``KafkaProducer`` and ``KafkaConsumer``,
 /// which is used to handle the connection to the Kafka ecosystem.
 final class RDKafkaClient: Sendable {
+    // Default size for Strings returned from C API
+    static let stringSize = 1024
+
     /// Determines if client is a producer or a consumer.
     enum ClientType {
         case producer
         case consumer
     }
 
-    // Default size for Strings returned from C API
-    static let stringSize = 1024
-
     /// Handle for the C library's Kafka instance.
     private let kafkaHandle: OpaquePointer
     /// A logger.
     private let logger: Logger
 
-    /// `librdkafka`'s main `rd_kafka_queue_t`.
-    private let mainQueue: OpaquePointer
+    /// `librdkafka`'s `rd_kafka_queue_t` that events are received on.
+    private let queue: OpaquePointer
 
     // Use factory method to initialize
     private init(
+        type: ClientType,
         kafkaHandle: OpaquePointer,
         logger: Logger
     ) {
         self.kafkaHandle = kafkaHandle
         self.logger = logger
 
-        self.mainQueue = rd_kafka_queue_get_main(self.kafkaHandle)
+        if type == .consumer {
+            if let consumerQueue = rd_kafka_queue_get_consumer(self.kafkaHandle) {
+                // (Important)
+                // Polling the queue counts as a consumer poll, and will reset the timer for `max.poll.interval.ms`.
+                self.queue = consumerQueue
+            } else {
+                fatalError("""
+                Internal error: failed to get consumer queue. \
+                A group.id should be set even when the client is not part of a consumer group. \
+                See https://github.com/edenhill/librdkafka/issues/3261 for more information.
+                """)
```

Comment:
Are we making sure this group id is set?

Reply:
So for our two consumption strategies:

```swift
public static func partition(
    _ partition: KafkaPartition,
    topic: String,
    offset: KafkaOffset = .end
)
        
public static func group(id groupID: String, topics: [String]) 
```

We assign the `group.id` ourselves for `.partition` and for `.group` the user supplies us with a `String`. However, when this 'groupID` `String` is `""` we run into the `fatalError`